### PR TITLE
Added extension to smarty to properly set up locale related settings

### PR DESCRIPTION
**Pull request by tboloo**
Disclaimer - this is not my original work, I merely copied it from https://code.google.com/p/smarty-gettext/
Anyway - with this additional file one is able to setup locale related stuff (textdomain, etc) from master smarty template via {locale path="{$smarty.current_dir}/PATH_TO_LOCALIZATION_FILES" domain="YOUR_TRANSLATION_DOMAIN(default 'messages'"}
If you find it useful - merge.

### CODE REVIEW

**Comment by glensc** [comment]
looks okay, but would you mind digging up more docs? what what does "stack=push" and "stack=pop" do, and how to use (i.e add more examples)

**Comment by tboloo** [comment]
OK, I cleaned the code, removed the stack operations and added tests, although I do not know how to make them pass the Travis CI build :/

**Comment by glensc** [comment]
meh. why you removed stack, i asked only to find and add docs for that, i think pushing and poping locale state is very useful.

as for travis, ignore errors from php 5.2, travis removed it's support (fixed already on master)

**Comment by glensc** [comment]
also, i prefer not storing binary content in git. should keep only source (.po) and rebuild if needed

File: function.locale.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * You should have received a copy of the GNU Lesser General Public          *
+ * License along with this library; if not, write to the Free Software       *
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA *
+ * ------------------------------------------------------------------------- *
+ *
+ * Installation: 
+ *
+ * @package smarty-gettext
+ * @link  http://code.google.com/p/smarty-gettext/
+ * @link  https://github.com/glensc/smarty-gettext/
```
Comment (glensc) [inline function.locale.php]: please use current project url here

**Comment by tboloo** [comment]
So I've changed the array syntax, updated project url. As far as *.mo files are concerned - tests would't pass without them, and compiling them before each run would require msgfmt to be installed and run, so I'll leave it up to your decision. As far as stack operation - I didn't find any useful documentation, so I came up with a test which shows how it might be used (more of a guess then real knowledge).
`$smarty->joined_template_dir` indeed would work for smarty3, `template_dir` should work for Smarty2 as well (haven't tested it though)

**Comment by glensc** [comment]
sorry, this was all too messy, so i cleaned it up, and squashed into one commit c12ac35

**Pull request closed without merging**